components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeMetaDataConsumer.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.xchange;
+
+import static org.apache.camel.component.xchange.XChangeConfiguration.METHOD_CURRENCIES;
+import static org.apache.camel.component.xchange.XChangeConfiguration.METHOD_CURRENCY_PAIRS;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.impl.DefaultConsumer;
+
+public class XChangeMetaDataConsumer extends DefaultConsumer {
+    
+    public XChangeMetaDataConsumer(XChangeEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+    }
+
+    @Override
+    public XChangeEndpoint getEndpoint() {
+        return (XChangeEndpoint) super.getEndpoint();
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        XChangeEndpoint endpoint = getEndpoint();
+        String method = endpoint.getConfiguration().getMethod();
+        
+        if (METHOD_CURRENCIES.equals(method)) {
```

Comment:
This looks really wrong as this consumer is run only once. Before it was scheduled so it calls that once every X period. Why did you change this? If its not scheduled then the 3rd party library must have some kind of event behavior or scheduling on its own that triggers and calls this consumer to process the message in a Camel route

Reply:
Also the routing should never process in the start method. Those are for starting up resources and whatnot. Only after its started it should poll/trigger on events and route messages - not before.

Reply:
The [ticker](https://github.com/tdiesler/camel/blob/CAMEL-12167/components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeMarketDataProducer.java#L48) functionality is now an ordinary producer. The user ask for the current price of a currency pair when needed (i.e. poll) instead of the component doing these polls on a scheduled basis.

Reply:
The [metadata consumer](https://github.com/tdiesler/camel/blob/CAMEL-12167/components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeMetaDataConsumer.java) is indeed something I'm not at all sure about. Here I want to give access to metadata associated with the underlying XChange (not the Camel Exchange). This data becomes available when the endpoint is created and does not change during its lifetime. The low level API is [on the endpoint](https://github.com/tdiesler/camel/blob/CAMEL-12167/components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeEndpoint.java#L95).

Reply:
Please see how the [test case](https://github.com/tdiesler/camel/blob/CAMEL-12167/components/camel-xchange/src/test/java/org/apache/camel/component/xchange/MetaDataConsumerTest.java) currently uses the metadata service